modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/rebalance/RebalanceRaftGroupEventsListener.java

```diff
@@ -308,13 +309,56 @@ private void doStableKeySwitch(
             Entry switchAppendEntry = values.get(switchAppendKey);
             Entry assignmentsChainEntry = values.get(assignmentsChainKey);
 
+            AssignmentsQueue pendingAssignmentsQueue = AssignmentsQueue.fromBytes(pendingEntry.value());
```

Comment:
This method is already fat enough... Let's maybe move this code to separate method, WDYT? Same for zone listener